src/main/java/com/squareup/javapoet/TypeSpec.java

```diff
@@ -98,6 +98,10 @@ public static Builder anonymousClassBuilder(String typeArgumentsFormat, Object..
         .build());
   }
 
+  public static Builder annotationBuilder(String name) {
```

Comment:
Can you add a checkState() in `addMethod` to make sure that we reject methods with default values unless the current typespec is an annotation? It's an easy check to add and makes the code a little easier to reason about.